src/ts/components/playbackspeedselectbox.ts

```diff
@@ -6,37 +6,60 @@ import {UIInstanceManager} from '../uimanager';
  * A select box providing a selection of different playback speeds.
  */
 export class PlaybackSpeedSelectBox extends SelectBox {
+  protected defaultPlaybackSpeeds: number[];
 
   constructor(config: ListSelectorConfig = {}) {
     super(config);
+    this.defaultPlaybackSpeeds = [0.25, 0.5, 1, 1.5, 2];
   }
 
   configure(player: bitmovin.PlayerAPI, uimanager: UIInstanceManager): void {
     super.configure(player, uimanager);
 
-    this.addItem('0.25', '0.25x');
-    this.addItem('0.5', '0.5x');
-    this.addItem('1', 'Normal');
-    this.addItem('1.5', '1.5x');
-    this.addItem('2', '2x');
-
+    this.addDefaultItems();
 
     this.onItemSelected.subscribe((sender: PlaybackSpeedSelectBox, value: string) => {
       player.setPlaybackSpeed(parseFloat(value));
       this.selectItem(value);
     });
 
-    let setDefaultValue = (): void => {
-      let playbackSpeed = String(player.getPlaybackSpeed());
-      if (!this.selectItem(playbackSpeed)) {
-        playbackSpeed = '1';
-        this.selectItem(playbackSpeed);
-      }
+    const setDefaultValue = (): void => {
+      const playbackSpeed = String(player.getPlaybackSpeed());
+      this.setSpeed(playbackSpeed);
     };
 
-    setDefaultValue();
-
-    // when the player hits onReady again, adjust the playback speed selection with fallback to default 1
+    // when the player hits onReady again, adjust the playback speed selection
     player.addEventHandler(player.EVENT.ON_READY, setDefaultValue);
+
+    if (player.EVENT.ON_PLAYBACK_SPEED_CHANGED) {
+      // Since player 7.7.0
+      player.addEventHandler(player.EVENT.ON_PLAYBACK_SPEED_CHANGED, setDefaultValue);
+    }
+  }
+
+  setSpeed(speed: string): void {
+    if (!this.selectItem(speed)) {
```

Comment:
I would handle the speeds as `number` everywhere except where it needs to be a string, i.e. convert it to `String` in `addItem` and `selectItem` and leave it `number` everywhere else. Because now it is `ON_PLAYBACK_SPEED_CHANGED` > `setDefaultValue` > `number` to `string` (`player.getPlaybackSpeed()`) > `setSpeed` > `string` to `number` (`addDefaultItems`) > `addDefaultItems` > `number` to `string` (`addItem`).